### ISSUE TITLE
[Bug] [C#] [Blazor] GrowthBook EvalFeature function always returns 'defaultValue' source 

### ISSUE THREAD

**Issue by eostlind** [growthbook/growthbook#2137]
### Summary

When executing the EvalFeature function of the GrowthBook object in the client runtime of a blazor application the returned value source is always 'defaultValue'

Using the following json feature returned from our SDK connection
```
{
    "status": 200,
    "features": {
        "test-message": {
            "defaultValue": true,
            "rules": [
                {
                    "force": false
                }
            ]
        }
    },
    "dateUpdated": "2024-02-16T21:18:21.099Z"
}
```

The 'force' rule is not applied. However, when running this exact same command in the server side application, the correct value is returned.

I suspect that an error is being thrown since the catch clause will return 'defaultValue', but Im not sure how to view the output of the logger associated with the GrowthBook class

#### Expected Behavior

growthBook.EvalFeature("test-message")

```
{
    "Value": false,
    "Source": "force"
}
```

#### Current Behavior

growthBook.EvalFeature("test-message")

```
{
    "Value": true,
    "Source": "defaultValue"
}
```

**Comment by Norhaven** [comment]
@eostlind Hi, logging would definitely be helpful here. Using the C# SDK with client-side Blazor, you could [inject an ILoggerFactory instance](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/logging?view=aspnetcore-8.0) into your client-side page and include that in the Context (with the `LoggerFactory` property) that you pass into GrowthBook, which will enable logging throughout the SDK. Client-side logs should show up in your browser's development tools console. Please try that and let me know if an error does occur. Thanks!

**Comment by eostlind** [comment]
@Norhaven Sorry for the late reply. It seems that the error occurs when using a Force rule and having the TrackingCallback field specified. The error I get is
```
fail: GrowthBook.GrowthBook[0]
      Encountered an unhandled exception while executing 'EvalFeature'
System.ArgumentNullException: Value cannot be null. (Parameter 'source')
   at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   at System.Linq.Enumerable.Any[TrackData](IEnumerable`1 source)
   at GrowthBook.GrowthBook.EvalFeature(String featureId)
```

I believe its caused because "Tracks" in the FeatureRule is null and there is no null-check
```
public FeatureResult EvalFeature(string featureId)
{
    ...
    if (_trackingCallback != null && item.Tracks.Any())
    ...
}
```

**Comment by Norhaven** [comment]
Hi @eostlind, thanks for tracking that down, I appreciate it! I have a [PR on the C# SDK repo](https://github.com/growthbook/growthbook-c-sharp/pull/33) now that should address this issue along with some others. Just wanted to give you a heads up on that. Thanks again!

**Comment by eostlind** [comment]
Thanks for the quick progress, im just waiting on a release which includes this change

**Comment by stevenm973** [comment]
@Norhaven Can you please confirm? I see that this says this was merged last week. However, we have not gotten an update here?

**Comment by Norhaven** [comment]
@eostlind @stevenm973 Thanks for your patience while we worked out some kinks with the release process! An updated version has been released on NuGet now, please upgrade to v1.0.5 and let us know how it goes. Thanks again!
